substrate/frame/support/src/traits/storage.rs

```diff
@@ -342,6 +342,80 @@ where
 
 impl_incrementable!(u8, u16, u32, u64, u128, i8, i16, i32, i64, i128);
 
+/// Wrap a type so that is `Drop` impl is never called.
+///
+/// Useful when storing types like `Imbalance` which would trigger their `Drop`
+/// implementation whenever they are written to storage as they are dropped after
+/// being serialized.
+#[derive(Default, Encode, Decode, DecodeWithMemTracking, MaxEncodedLen, TypeInfo)]
+pub struct NoDrop<T: Default>(T);
+
+impl<T: Default> Drop for NoDrop<T> {
+	fn drop(&mut self) {
+		mem::forget(mem::take(&mut self.0))
+	}
+}
+
+/// Sealed trait that marks types with suppressed Drop implementations.
+///
+/// Usefull for constraining your storage items types by this bound to make
+/// sure they wont't runt Drop when stored.
+pub trait SuppressedDrop: sealed::Sealed {
+	/// The wrapped whoe drop function is ignored,
+	type Inner;
+
+	fn new(inner: Self::Inner) -> Self;
+	fn as_ref(&self) -> &Self::Inner;
+	fn as_mut(&mut self) -> &mut Self::Inner;
+	fn into_inner(self) -> Self::Inner;
+}
+
+impl SuppressedDrop for () {
+	type Inner = ();
+
+	fn new(inner: Self::Inner) -> Self {
+		inner
+	}
+
+	fn as_ref(&self) -> &Self::Inner {
+		self
+	}
+
+	fn as_mut(&mut self) -> &mut Self::Inner {
+		self
+	}
+
+	fn into_inner(self) -> Self::Inner {
+		self
+	}
+}
+
+impl<T: Default> SuppressedDrop for NoDrop<T> {
+	type Inner = T;
+
+	fn as_ref(&self) -> &Self::Inner {
+		&self.0
+	}
+
+	fn as_mut(&mut self) -> &mut Self::Inner {
+		&mut self.0
+	}
+
+	fn into_inner(mut self) -> Self::Inner {
+		mem::take(&mut self.0)
```

Comment:
```suggestion
		self.0
```

Reply:
Can't move fields of types that implement `Drop`.